### PR TITLE
Refactored user type to criteria type in onCriteria tests

### PR DESCRIPTION
**Pull request by xstrengthofonex**

### CODE REVIEW

File: internal/co_coders/criteria/criteria.go [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package criteria
 
+
+
```
Comment (unional) [inline internal/co_coders/criteria/criteria.go:4]: extra lines?

File: internal/co_coders/match/on_collab_style_test.go [hunk 2]
```diff
@@ -40,34 +40,33 @@ func TestMatchOnCollabStyleSubsetReturnsAMatch(t *testing.T) {
 }
 
 func TestMatchHasAScore(t *testing.T) {
-	aCriteria := user.New([]criteria.CollabStyle{criteria.Team, criteria.Pair})
-	users := []user.User{aCriteria}
-	aMatch := OnCriteria(aCriteria, users)
+	matchCriteria := criteria.Criteria{
+		CollabStyles: []criteria.CollabStyle{criteria.Team}}
+	allMatchUser := user.New(matchCriteria.CollabStyles)
```
Comment (akessner) [inline internal/co_coders/match/on_collab_style_test.go:45]: Why not just pass in matchCriteria to the user constructor?

**Review by akessner — COMMENTED** [review]
I think user should only contain the criteria object and not the specifics.

Or are you trying to keep the refactoring small?

File: internal/co_coders/match/on_collab_style_test.go (hunk 2 above)
Comment (unional) [reply internal/co_coders/match/on_collab_style_test.go:45]: Agree, it should be criteria. User should not know about details like CollabStyles.

**Comment by xstrengthofonex** [comment]
Yeah, I was trying to keep the refactoring smaller. I'll do it in the next PR. I realized it involves changes in the other tests and there a few other things I'd like to refactor with it.

File: internal/co_coders/match/on_collab_style_test.go (hunk 2 above)
Comment (xstrengthofonex) [reply internal/co_coders/match/on_collab_style_test.go:45]: I'll refactor it in the next PR

**Review by akessner — APPROVED** [review]

**Review by unional — APPROVED** [review]

**Pull request merged**